File: lib/src/main/java/com/orientsec/easysocket/utils/AndroidLogger.java
package com.orientsec.easysocket.utils;

import android.util.Log;

/**
 * AndroidLogger is a utility class that implements the Logger interface for logging messages
 * on the Android platform. It provides methods to log messages at different levels (error,
 * info, warning, debug) and supports detailed logging with thread information and a custom suffix.
 */
class AndroidLogger implements Logger {
    // The default tag used for logging
    private static final String TAG = "EasySocket";

    // A custom suffix appended to log messages
    private final String suffix;

    // Minimum log level to output
    private final int minLogLevel;

    /**
     * Constructs an AndroidLogger instance with the specified options and suffix.
     *
     * @param suffix      A custom suffix to append to log messages.
     * @param minLogLevel The minimum log level to output (e.g., Log.DEBUG, Log.INFO).
     */
    public AndroidLogger(String suffix, int minLogLevel) {
        this.suffix = suffix;
        this.minLogLevel = minLogLevel;
    }

    /**
     * Formats the log message by appending the suffix and thread information if detailed logging is enabled.
     *
     * @param msg The original log message.
     * @return The formatted log message.
     */
    private String formatMsg(String msg) {
        return msg + suffix +
                "  Thread:[" + Thread.currentThread().getName() + "]";
    }

    /**
     * Checks if the log level is enabled for output.
     *
     * @param level The log level to check.
     * @return True if the log level is enabled, false otherwise.
     */
    private boolean isIgnored(int level) {
        return level < minLogLevel;
    }

    @Override
    public void e(String msg) {
        if (isIgnored(Log.ERROR)) return;
        Log.e(TAG, formatMsg(msg));
    }

    @Override
    public void e(String msg, Throwable t) {
        if (isIgnored(Log.ERROR)) return;
        Log.e(TAG, formatMsg(msg), t);
    }

    @Override
    public void i(String msg) {
        if (isIgnored(Log.INFO)) return;
        Log.i(TAG, formatMsg(msg));
    }

    @Override
    public void i(String msg, Throwable t) {
        if (isIgnored(Log.INFO)) return;
        Log.i(TAG, formatMsg(msg), t);
    }

    @Override
    public void w(String msg) {
        if (isIgnored(Log.WARN)) return;
        Log.w(TAG, formatMsg(msg));
    }

    @Override
    public void w(String msg, Throwable t) {
        if (isIgnored(Log.WARN)) return;
        Log.w(TAG, formatMsg(msg), t);
    }

    @Override
    public void d(String msg) {
        if (isIgnored(Log.DEBUG)) return;
        Log.d(TAG, formatMsg(msg));
    }

    @Override
    public void d(String msg, Throwable t) {
        if (isIgnored(Log.DEBUG)) return;
        Log.d(TAG, formatMsg(msg), t);
    }
}

File: lib/src/androidMain/kotlin/com/orientsec/easysocket/utils/AndroidLogger.kt
package com.orientsec.easysocket.utils

import android.util.Log

/**
 * AndroidLogger is a utility class that implements the Logger interface for logging messages
 * on the Android platform. It provides methods to log messages at different levels (error,
 * info, warning, debug) and supports detailed logging with thread information and a custom suffix.
 */
internal class AndroidLogger(
    private val suffix: String,
    private val minLogLevel: Int
) : Logger {

    /**
     * Formats the log message by appending the suffix and thread information if detailed logging is enabled.
     *
     * @param msg The original log message.
     * @return The formatted log message.
     */
    private fun formatMsg(msg: String): String {
        return msg + suffix + "  Thread:[" + Thread.currentThread().name + "]"
    }

    /**
     * Checks if the log level is enabled for output.
     *
     * @param level The log level to check.
     * @return True if the log level is enabled, false otherwise.
     */
    private fun isIgnored(level: Int): Boolean {
        return level < minLogLevel
    }

    override fun e(msg: String) {
        if (isIgnored(Log.ERROR)) return
        Log.e(TAG, formatMsg(msg))
    }

    override fun e(msg: String, t: Throwable?) {
        if (isIgnored(Log.ERROR)) return
        Log.e(TAG, formatMsg(msg), t)
    }

    override fun i(msg: String) {
        if (isIgnored(Log.INFO)) return
        Log.i(TAG, formatMsg(msg))
    }

    override fun i(msg: String, t: Throwable?) {
        if (isIgnored(Log.INFO)) return
        Log.i(TAG, formatMsg(msg), t)
    }

    override fun w(msg: String) {
        if (isIgnored(Log.WARN)) return
        Log.w(TAG, formatMsg(msg))
    }

    override fun w(msg: String, t: Throwable?) {
        if (isIgnored(Log.WARN)) return
        Log.w(TAG, formatMsg(msg), t)
    }

    override fun d(msg: String) {
        if (isIgnored(Log.DEBUG)) return
        Log.d(TAG, formatMsg(msg))
    }

    override fun d(msg: String, t: Throwable?) {
        if (isIgnored(Log.DEBUG)) return
        Log.d(TAG, formatMsg(msg), t)
    }

    companion object {
        // The default tag used for logging
        private const val TAG = "EasySocket"
    }
}
